### PR TITLE
Dynamic PORT selection for PFC storm testing

### PR DESCRIPTION
**Pull request by maunnikr-cisco**
### Description of PR

In the test file : telemetry/events/swss_events.py
in the function generate_pfc_storm, in the line
queue_oid = duthost.get_queue_oid(PFC_STORM_TEST_PORT, PFC_STORM_TEST_QUEUE)
The PFC_STORM_TEST_PORT - the port chosen to test the PFC storm on - is assigned a fixed value of “Ethernet4”, whereas, some DUTs don't have a port Ethernet4 associated with it’s ASIC instance. Therefore this test is passing when assigning a value of Ethernet8 or Ethernet16 which are valid ports.

Changing fixed assigned value, replacing it with a dynamically selected port from the list of ports that have both open and admin status as "UP"

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [x] 202305

#### What is the motivation for this PR?
Want to ensure that telemetry/test_events.py test case passes - and that when selecting a port to do a PFC storm on, we are chasing a valid port, instead of a fixed one. 

#### How did you verify/test it?
Successfully ran telemetry/test_events.py

#### Supported testbed topology if it's
 a new test case?

### Documentation

Logs: 
[test_events.log](https://github.com/sonic-net/sonic-mgmt/files/13635360/test_events.log)
[test_events_master.log](https://github.com/sonic-net/sonic-mgmt/files/14642407/test_events_master.log)

### CODE REVIEW

**Comment by Aravind-Subbaroyan** [comment]
@kevinskwang  - could you review this PR?

**Comment by developfast** [comment]
@maunnikr-cisco can you please paste the test output after the change in the pr description?

**Review by developfast — COMMENTED** [review]
Just a bump on adding test output to the PR description, otherwise lgtm

**Comment by developfast** [comment]
@maunnikr-cisco the logs even in debug mode don't seem to note `Generating pfc storm` on line 46 - are you certain that function is called?

**Comment by maunnikr-cisco** [comment]
> @maunnikr-cisco the logs even in debug mode don't seem to note `Generating pfc storm` on line 46 - are you certain that function is called?

I have updated the logs, and the code with some improvements - the new logo file shows the "generate_pfc_storm" function being called at line 8412

**Review by developfast — APPROVED** [review]
lgtm

**Comment by maunnikr-cisco** [comment]
@kevinskwang @XuChen-MSFT @wangxin this needs approval from someone with write and merge access to be merged. Please take a look.

**Comment by kevinskwang** [comment]
Could you upload the test results for both master and 202305?

**Review by bpar9 — APPROVED** [review]

**Comment by bpar9** [comment]
@maunnikr-cisco can you please upload the required logs asap

**Comment by maunnikr-cisco** [comment]
@kevinskwang @bpar9 have uploaded logs for master and 20

**Pull request merged**
